### ISSUE TITLE
how did you manage the panfrost to work?

### ISSUE THREAD

**Issue by cuu** [avafinger/orangepi-h6-ubuntu-base-minimal#7]
hi avafinger

Did you build the mesa and libdrm manually, not using them from 18.04 repo?

thanks

**Comment by cuu** [comment]
Did you build the mesa and libdrm manually, not using them from 18.04 repo?

**Comment by avafinger** [comment]
Yes. Panfrost in this kernel was in its early stage. Not stable for a good desktop experience, unfortunately.

**Comment by cuu** [comment]
and I found out that panfrost for h6 t720 is not stable except I did this:

```
cd /sys/devices/platform/soc/1800000.gpu/devfreq/1800000.gpu/
echo userspace > governor
echo 756000000 > max_freq
echo 756000000 > min_freq
```

Did you do that in your panfrost testing, too?

**Comment by avafinger** [comment]
No, i did not. Please, report your findings. Thanks.

**Comment by cuu** [comment]
I got that trick from https://forum.armbian.com/topic/12516-unable-to-make-panfrost-work-on-h6/page/4/

and I think you are so lucky, if I did not set those settings when I run glmark2, the X will die in a very short time
even I send a reboot from ssh  to the Linux, the system still hangs

I used the 
https://archive.armbian.com/orangepioneplus/archive/Armbian_20.08_Orangepioneplus_bullseye_current_5.7.15.img.xz

And for your minimal  image, I tried once, I need to rename /dev/dri/card1  to be other name else so that the X can show up, or 
x will fail with an error 
```
Fatal server error:
[    23.406] (EE) AddScreen/ScreenInit failed for driver 0
```

I was very curious that what exactly os image you were testing the H6 wth panfrost enabled,
it is so unique that you don't need those tricks above(setting the freqs)

If it is convenient for you, would you like to upload a working os image included your kernel?

best regards
